snuba/datasets/outcomes_processor.py

```diff
@@ -18,14 +18,37 @@
 metrics = MetricsWrapper(environment.metrics, "outcomes.processor")
 
 
+OUTCOME_ABUSE = 4
+OUTCOME_CLIENT_DISCARD = 5
+
+CLIENT_DISCARD_REASONS = frozenset([
```

Comment:
These discard reasons were identified in the DACI for the feature. We will likely adapt these once the actual SDK implementations land.

Reply:
I think it would be more desirable to validate those discard reasons in internal relay and have snuba accept arbitrary strings. It does not necessarily mean that external relays do the same validation. I think that would be more in line with how we generally do schema validation. If an SDK starts blasting invalid SDK reasons it would be more desirable to drop them in pops than impose them on every outcomes consumer.

Reply:
We decided against this with Jan because then we need a mechanism to communicate these to customer relays as they might not be updated.

Reply:
I would propose to have customer relays pass this through and actually validate them only in internal relays

Reply:
decided on keeping this in a discussion